pkg/codegen/go/gen_test.go

```diff
@@ -124,21 +125,25 @@ func TestEnumUsage(t *testing.T) {
 				Type: tree.RubberTreeVarietyRuby,
 			})
 			require.NoError(t, err)
-			require.NotNil(t, tree)
+			require.NotNil(t, rubberTree)
 			var wg sync.WaitGroup
 			wg.Add(1)
 			pulumi.All(
-				tree.URN(), tree.Container.Material(), tree.Container.Color(), tree.Container.Size(), tree.Type,
+				rubberTree.URN(),
+				rubberTree.Container.Material(),
+				rubberTree.Container.Color(),
+				rubberTree.Container.Size(),
+				rubberTree.Type,
 			).ApplyT(func(all []interface{}) error {
 				urn := all[0].(pulumi.URN)
 				material := all[1].(*string)
 				color := all[2].(*string)
-				size := all[3].(*int)
-				typ := all[4].(string)
+				size := all[3].(*plant.ContainerSize)
+				typ := all[4].(tree.RubberTreeVariety)
 				assert.Equal(t, *material, "ceramic", "unexpected material on resource: %v", urn)
 				assert.Equal(t, *color, "Magenta", "unexpected color on resource: %v", urn)
-				assert.Equal(t, *size, 22, "unexpected size on resource: %v", urn)
-				assert.Equal(t, typ, "Ruby", "unexpected type on resource: %v", urn)
+				assert.Equal(t, *size, plant.ContainerSize(22), "unexpected size on resource: %v", urn)
+				assert.Equal(t, typ, tree.RubberTreeVarietyRuby, "unexpected type on resource: %v", urn)
```

Comment:
Note that it is no longer possible to pass in untyped strings/integers to strict enums. This is in line with how strings must be wrapped in `pulumi.String` to function correctly.